packages/dev/s2-docs/src/client.tsx

```diff
@@ -30,14 +30,52 @@ let updateRoot = hydrate({
 let currentNavigationId = 0;
 let currentAbortController: AbortController | null = null;
 
+// Store scroll positions by pathname
+const scrollPositions = new Map<string, {scrollTop: number, windowScrollTop: number}>();
```

Comment:
Maybe we can save these in the history entry itself? https://stackoverflow.com/a/16732655

Might also have to turn off the browser's defaults? https://developer.chrome.com/blog/history-api-scroll-restoration

I did wonder if the defaults would work if we moved the `key` from the `<main>` element further down (e.g. article) so that it doesn't get replaced on each navigation. Did you try that?